src/devtoolsPanel.ts

```diff
@@ -311,35 +312,42 @@ export class DevToolsPanel {
         // Convert the local url to a workspace path
         const transformer = new debugCore.UrlPathTransformer();
         void transformer.launch({ pathMapping: this.config.pathMapping });
-        const localSource = { path: sourcePath };
-        await transformer.fixSource(localSource);
 
-        sourcePath = localSource.path || sourcePath;
+        // origin in this case is trivial since we expect fixSource to take it out
+        const localSource = { path: sourcePath, origin: 'vscode-webview://' };
+        await transformer.fixSource(localSource);
 
-        // Convert the workspace path into a VS Code url
-        let uri: vscode.Uri | undefined;
-        try {
-            if (vscode.env.remoteName) {
-                uri = vscode.Uri.parse(sourcePath, true);
-            } else {
-                uri = vscode.Uri.file(sourcePath);
-            }
-        } catch {
-            uri = undefined;
+        // per documentation if the file was correctly resolved origin will be cleared.
+        // https://github.com/Microsoft/vscode-chrome-debug-core/blob/main/src/transformers/urlPathTransformer.ts
+        if (!localSource.origin) {
+            // Convert the workspace path into a VS Code url
+            const uri = vscode.Uri.file(localSource.path);
+            await this.openInEditor(uri, line, column);
+        } else {
+            await ErrorReporter.showInformationDialog({
+                errorCode: ErrorCodes.Error,
+                title: 'Error while opening file in editor.',
+                message: `${sourcePath} file is not mapped to a local file.`,
+            });
         }
+    }
 
-        // Finally open the document if it exists
-        if (uri) {
+    private async openInEditor(uri:vscode.Uri, line: number, column: number){
```

Comment:
nit: I'm not sure why this was broken out into another function, unless you use it somewhere else now?

Reply:
I was using it in different places in a previous iteration but I left it as it helps the testing.